File: frontend/types/campaign.ts
export interface Campaign {
  id: string;

  merchant_id: string;

  title: string;

  description: string;

  audience: string;

  discount_percentage: number;

  expected_revenue: number;

  confidence: number;

  status: string;

  requires_approval: boolean;
}

export interface CampaignListResponse {
  campaigns: Campaign[];

  total: number;
}

File: backend/app/schemas/campaign.py
from uuid import UUID

from pydantic import BaseModel
from pydantic import ConfigDict


class CampaignResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: UUID

    merchant_id: UUID

    title: str

    description: str

    audience: str

    discount_percentage: int

    expected_revenue: float

    confidence: int

    status: str

    requires_approval: bool


class CampaignListResponse(BaseModel):
    campaigns: list[CampaignResponse]

    total: int